### ISSUE TITLE
[Auth] Constant ~5% CPU usage on Windows

### ISSUE THREAD

**Issue by mind12** [ente-io/ente#3031]
### Description

Whether the window is hidden or visible, Ente Auth is always at between 3% to 5% CPU usage on an 5800H 16 threads mobile processor using Windows 10. 
This is unacceptably high for the task it should do and affects battery life badly.

(Not to mention the constant 15% GPU usage referenced here: https://github.com/ente-io/ente/issues/2686 )

![image](https://github.com/user-attachments/assets/e11b4bc9-67f1-498e-a046-e35a829ee753)

### Version

3.1.3

### What product are you using?

Ente Auth

### What platform are you using?

Desktop - Windows

**Comment by bryanparadis** [comment]
It seems to be high usage on all 3 desktop platforms. I can verify that it auto hides onto systray on PopOS and then adds 5W of battery drain on my Framework 13.

**Comment by ua741** [comment]
@mind12 Can you please confirm if you see improvement in the latest version?

**Comment by mind12** [comment]
> @mind12 Can you please confirm if you see improvement in the latest version?

Sadly no improvements, it's still constantly hogging the CPU by 4-5%.
Even if I minimize it to the taskbar. 
Let me know how I can help to fix this.

![image](https://github.com/user-attachments/assets/b3e78ae7-6213-410e-a79b-f1c00952ea10)

**Comment by angelosc** [comment]
I have the exact same issue

**Comment by weikai** [comment]
I have about 38 accounts in Ente Auth. It's taking up 25% cpu usage on Mac Mini 4. I think the UI should be redesigned or the code should not be generated until the accounts are showing up in the screen.

**Comment by bryanparadis** [comment]
> I have about 38 accounts in Ente Auth. It's taking up 25% cpu usage on Mac Mini 4. I think the UI should be redesigned or the code should not be generated until the accounts are showing up in the screen.

How is it now? I just checked fresh update and % of cpu usage change with how many codes I have visible/size of window. With only 1 code it's basically 0. With a fullscreen of codes it's about 5%.

**Comment by bryanparadis** [comment]
> > > I have about 38 accounts in Ente Auth. It's taking up 25% cpu usage on Mac Mini 4. I think the UI should be redesigned or the code should not be generated until the accounts are showing up in the screen.
> > 
> > 
> > Is that with v4.1.0? On Windows at least the usage now is 0% with one code visible and only 5% fullscreen with tons of codes.
> 
> I updated to 4.1.0 and it's still using 3-4% CPU with the window open or on the tray. I have about 15 entries.

On Windows?

**Comment by mind12** [comment]
> > > > I have about 38 accounts in Ente Auth. It's taking up 25% cpu usage on Mac Mini 4. I think the UI should be redesigned or the code should not be generated until the accounts are showing up in the screen.
> > > 
> > > 
> > > Is that with v4.1.0? On Windows at least the usage now is 0% with one code visible and only 5% fullscreen with tons of codes.
> > 
> > 
> > I updated to 4.1.0 and it's still using 3-4% CPU with the window open or on the tray. I have about 15 entries.
> 
> On Windows?

yes, as if you check, I opened this issue on a win 10.

**Comment by bryanparadis** [comment]
> > > > > I have about 38 accounts in Ente Auth. It's taking up 25% cpu usage on Mac Mini 4. I think the UI should be redesigned or the code should not be generated until the accounts are showing up in the screen.
> > > > 
> > > > 
> > > > Is that with v4.1.0? On Windows at least the usage now is 0% with one code visible and only 5% fullscreen with tons of codes.
> > > 
> > > 
> > > I updated to 4.1.0 and it's still using 3-4% CPU with the window open or on the tray. I have about 15 entries.
> > 
> > 
> > On Windows?
> 
> yes, as if you check, I opened this issue on a win 10.

 I have 23 codes full screen and if I hit minimize it drops down to 0% on the tray most of the time with small spikes up to 2% once in a while. It's definitely not chugging 23 codes all the time. It doesn't drop down instantly though. There still could be some optimization done but on my machine it is night and day difference from fullscreen to minimized to tray. I am on Windows 11 though.

**Comment by degritsenko** [comment]
<img width="461" alt="Screenshot 2024-11-29 at 17 05 29" src="https://github.com/user-attachments/assets/9731d3d8-d9a2-4161-8306-094fae54d78d">
macbook air m1

**Comment by aith85** [comment]
Same on W10 22H2 19045.5198.
Ente v1.4.5 and previous one.
Constant 25-30% cpu usage when open.

![image](https://github.com/user-attachments/assets/a9e8d1a4-d0b3-42ec-a734-789bf5f0c63f)

**Comment by tn-mmyers** [comment]
Actually caught this on a low CPU usage - but the constant ~20% or so of GPU usage is a bit much... 

![image](https://github.com/user-attachments/assets/5dd1647b-a59e-4a49-b030-9df6ec8f049e)

**Comment by mind12** [comment]
4.1.6 has the same usage for me as before or even higher. 
I close the app once I logged in to a service.

**Comment by aith85** [comment]
> Actually caught this on a low CPU usage - but the constant ~20% or so of GPU usage is a bit much... 4.1.5 btw
> 
> ![image](https://private-user-images.githubusercontent.com/191257755/394319488-5dd1647b-a59e-4a49-b030-9df6ec8f049e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM5NTU0OTQsIm5iZiI6MTczMzk1NTE5NCwicGF0aCI6Ii8xOTEyNTc3NTUvMzk0MzE5NDg4LTVkZDE2NDdiLWE1OWUtNGE0OS1iMDMwLTlkZjZlYzhmMDQ5ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIxMVQyMjEzMTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jM2NlN2NiYzk3MjE4MTUwZGM3MTM1YTc3YWZiYWYxMDE0NDFlODdlMWIwNWQ5MzA2ZTJkNGVjOTZlNWEyNzA4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Pa2YSQ5PJa7kukna5C4BcRA9ZSApHHNnpLwk2HSA8kM)

**Comment by aith85** [comment]
On the contrary, v4.1.6 on W11 22631.4602:
0% CPU but constant 10~12% GPU
![image](https://github.com/user-attachments/assets/b97b1a7c-d01e-41b2-8a08-5699272942f1)

**Comment by ua741** [comment]
Does the CPU usage on Windows remains high when no code is visible on the UI? This can be tested by either adding app lock or searching with a keyword that results in no match.

**Comment by ua741** [comment]
Please upgrade to [v4.2.1.](https://github.com/ente-io/ente/releases/tag/auth-v4.2.1) 
Going forward, we will track high CPU issue [here](https://github.com/ente-io/ente/issues/2003#issuecomment-2563380828).

**Comment by aith85** [comment]
Upgraded to 4.2.2.

I can see the timer bar animation is not fluid anymore, and the CPU usage is 0~1% now.

However I've noticed another issue now.
I've got many services saved, and when I scroll down and up again, services that went hidden on the first scroll and came visible again have the timer bar reset to 100%, but it's not synchronized with the code expiration.
Tried to check against my android app: when at 50% I scrolled out the service and went back to it, and on windows the bar started from 100% again, while on my phone continued from 50% to 0%.
Hope it makes sense, as I'm not sure on how to make a video graying out all sensitive info.

Thanks
